examples/sample_test_models/model_rotxx_test.py

```diff
@@ -0,0 +1,14 @@
+# Input a number for rotation and a string via file
+def rotxx_func(x, txt):
+    alphabet = "abcdefghijklmnopqrstuvwxyz"
+    rotxx = ""
+    txt_string = txt.decode("utf-8")
```

Comment:
put the `.lower` here, since you unconditionally call it on every character

Reply:
You can also do a filter here, so you don't have to call `in alphabet` below:

`txt_string = "".join(_ for _ in txt.decode("utf-8").lower() if _ in string.ascii_lowercase)`

then below, `rotxx = "".join(string.ascii_lowercase[(ascii_lowercase.find(_) + x) % 26] for _ in txt_string)`

There are other clever ways to do this using string translation, but this should work.

Reply:
The only "issue" with this is if there is a character that isn't in `ascii.lowercase` such as a space, period, etc. it no longer keeps the formatted spaces or punctuation points (and does not hold the capitalization of letters, hencewhy it was called on every character rather than lowering the string). If this doesn't matter, then I'd be happy to put in these changes. If it does then I can try to adjust for those situations